plugins/kn-migration/cmd/migrate.go

```diff
@@ -0,0 +1,290 @@
+		err = migrationclientD.PrintServiceWithRevisions("destination")
+		if err != nil {
+			fmt.Printf(err.Error())
+			os.Exit(1)
+		}
+
+		fmt.Println("\nNow migrate all Knative service resources:")
+		fmt.Println("From the source", color.BlueString(namespaceS), "namespace of cluster", color.CyanString(kubeconfigS))
+		fmt.Println("To the destination", color.BlueString(namespaceD), "namespace of cluster", color.CyanString(kubeconfigD))
+
+		cfgD, err := clientcmd.BuildConfigFromFlags("", kubeconfigS)
+		clientset, err := clientset.NewForConfig(cfgD)
+		if err != nil {
+			fmt.Printf(err.Error())
+			os.Exit(1)
+		}
+		namespaceExists, err := namespaceExists(*clientset, namespaceD)
+		if err != nil {
+			fmt.Printf(err.Error())
+			os.Exit(1)
+		}
+
+		if !namespaceExists {
+			fmt.Println("Create namespace", color.BlueString(namespaceD), "in destination cluster")
+			nsSpec := &apiv1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceD}}
+			_, err = clientset.CoreV1().Namespaces().Create(nsSpec)
+			if err != nil {
+				fmt.Printf(err.Error())
+				os.Exit(1)
+			}
+		} else {
+			fmt.Println("Namespace", namespaceD, "already exists in destination cluster")
+		}
+		if err != nil {
+			fmt.Printf(err.Error())
+			os.Exit(1)
+		}
+
+		servicesS, err := migrationclientS.ListService()
+		if err != nil {
+			fmt.Printf(err.Error())
+			os.Exit(1)
+		}
+		for i := 0; i < len(servicesS.Items); i++ {
+			serviceS := servicesS.Items[i]
+			serviceExists, err := migrationclientD.ServiceExists(serviceS.Name)
+			if err != nil {
+				fmt.Printf(err.Error())
+				os.Exit(1)
+			}
+
+			if serviceExists {
+				if cmd.Flag("force").Value.String() == "false" {
+					fmt.Println("Cannot migrate service", color.CyanString(serviceS.Name), "in namespace", color.BlueString(namespaceS),
+						"because the service already exists and no --force option was given\n")
+					os.Exit(1)
+				}
+				fmt.Println("Deleting service", color.CyanString(serviceS.Name), "from the destination cluster and recreate as replacement")
+				migrationclientD.DeleteService(serviceS.Name)
+				if err != nil {
+					fmt.Printf(err.Error())
+					os.Exit(1)
+				}
+			}
+
+			_, err = migrationclientD.CreateService(&serviceS)
+			if err != nil {
+				fmt.Printf(err.Error())
+				os.Exit(1)
+			}
+			fmt.Println("Migrated service", color.CyanString(serviceS.Name), "Successfully")
+
+			serviceD, err := migrationclientD.GetService(serviceS.Name)
+			if err != nil {
+				fmt.Printf(err.Error())
+				os.Exit(1)
+			}
+
+			config, err := migrationclientD.GetConfig(serviceD.Name)
+			if err != nil {
+				fmt.Printf(err.Error())
+				os.Exit(1)
+			}
+			config_uuid := config.UID
+
+			revisionsS, err := migrationclientS.ListRevisionByService(serviceS.Name)
+			if err != nil {
+				fmt.Printf(err.Error())
+				os.Exit(1)
+			}
+			for i := 0; i < len(revisionsS.Items); i++ {
+				revisionS := revisionsS.Items[i]
+				if revisionS.Name != serviceS.Status.LatestReadyRevisionName {
```

Comment:
@zhangtbj will this check not be true always in the below scenario of traffic split ?
```
  traffic:
  - latestRevision: false
    percent: 50
    revisionName: envtest-v1
  - latestRevision: false
    percent: 50
    revisionName: envtest-v2
```

Reply:
Let me try this traffic split case.

Thanks!

Reply:
I have verified, after the migration, the traffic is kept and same as before:

This is the ksvc from source cluster:
```
  traffic:
  - latestRevision: false
    percent: 50
    revisionName: echo-vqhwd-1
  - latestRevision: false
    percent: 50
    revisionName: echo-dlmxf-2
```

And this is the ksvc traffic from dest cluster:
```
  traffic:
  - latestRevision: false
    percent: 50
    revisionName: echo-vqhwd-1
  - latestRevision: false
    percent: 50
    revisionName: echo-dlmxf-2
```

Because all info and configrations are kept in the service resource, so we won't lost them, you can see the **revision name** are also totally same.

Thanks! :)